docs/application-flow.rst

```diff
@@ -0,0 +1,402 @@
+
+.. _find-equilibrium:
+
+Find market equilibrium
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the main part of MUSE, in which the agents in the different sectors will invest in new - or old - technologies to make sure that the supply of commodities matches their demand in years to come across all the regions of the simulation.
+
+An overall picture of this process can be seen in the following chart, but there are many fine-grained steps related to specific objectives and criteria that heavily influence the results of the calculation. These steps are described in other parts of the documentation.
+
+.. graphviz::
+    :align: center
+    :alt: Main loop to find the market equilibrium
+
+    digraph find_equilibrium {
+            fontname="Helvetica,Arial,sans-serif"
+            node [fontname="Helvetica,Arial,sans-serif", shape=box, style=rounded]
+            edge [fontname="Helvetica,Arial,sans-serif"]
+            rankdir=LR
+            clusterrank=local
+            newrank=true
+
+            {node [shape=""]; start; end;}
+            exclude [label="Exclude\ncommodities\nfrom market"];
+            single_year [label="Single year\niteration", fillcolor="lightgrey", style="rounded,filled"]
+            maxiter [label="Maxium iter?", shape=diamond, style=""]
+            converged [label="Converged?", shape=diamond, style=""]
+            prices [label="Update with\nconverged prices"]
+            {node [label="Update with not\nconverged prices"]; prices1; prices2;}
+            check_demand [label="Check demand\nfulfilment"]
+            equilibrium [label="Check\nequilibrium"]
+
+            start -> exclude -> single_year -> equilibrium -> check_demand -> converged
+            converged -> prices [label="Yes"]
+            prices -> end
+            converged -> maxiter [label="No"]
+            maxiter -> prices2 [label="No"]
+            maxiter -> prices1 [label="Yes"]
+            prices1 -> end
+            prices2 -> single_year [constraint=false]
+            {rank=same; prices2; single_year}
+            {rank=same; maxiter; converged;}
+            {rank=same; prices1; prices;}
+        }
+
+Single year iteration
+~~~~~~~~~~~~~~~~~~~~~
+
+Both in the carbon budget and in the equilibrium calculation, a single year iteration step is involved. It is in this step where MUSE will go through each sector and use the agents to appropriately invest in different technologies, aiming to match these two factors.
+
+**As sectors have different priorities, sectors with lower priorities will run last and see a market updated by the higher priority sectors**. In general, demand sectors should run before conversion sectors and these before supply sectors, such that the later can see the real demand. As these investment in the supply sectors will change the prices of their commodities, the demand of those commodities will change. Balancing that is the purpose of the :ref:`find-equilibrium` loop described above.
```

Comment:
One point to clarify on the market, what updates at every sector execution is production and consumption. Prices are updated at end, once all the sectors have iterated.